crmd/remote_lrmd_ra.c

```diff
@@ -1173,4 +1189,43 @@ remote_ra_process_pseudo(xmlNode *xml)
         }
     }
     freeXpathObject(search);
+
+    /* if we get a list of nodes in maintenance with the transition
+     * we clear maintenance in all remote-nodes and just
+     * set it for the nodes given
+     */
+    search = xpath_search(xml, XPATH_PSEUDO_MAINTENANCE_NODES);
+    crm_trace("Transition holds %d nodes in maintenance-mode",
+              numXpathResults(search));
+    if (numXpathResults(search)) {
+        lrm_state_t *lrm_state;
+        int result;
+
+        lrm_state_foreach_remote(reset_maintenance, NULL);
+
+        for (result = 0; result < numXpathResults(search); result++) {
+            lrm_state = lrm_state_find(ID(getXpathResult(search, result)));
+
+            if (lrm_state && lrm_state->remote_ra_data) {
+                remote_ra_maintenance(lrm_state, TRUE);
+            }
+        }
+    }
+    freeXpathObject(search);
 }
+
+void
+remote_ra_maintenance(lrm_state_t * lrm_state, gboolean maintenance)
+{
+    remote_ra_data_t *ra_data = lrm_state->remote_ra_data;
+
+    ra_data->is_maintenance = maintenance;
+}
+
+gboolean
+remote_ra_is_in_maintenance(lrm_state_t * lrm_state)
+{
+    remote_ra_data_t *ra_data = lrm_state->remote_ra_data;
+
+    return ra_data->is_maintenance;
+}
```

Comment:
but then we would have to reveal remote_ra_data_t in the header as well ...
performance-critical is probably anyway just the calling of remote_ra_maintenance from
remote_lrmd_ra.c that probably is gonna be inlined automatically by the compiler anyway.
I don't know in how far usage of 'inline' in the header without definition and 'extern inline'
with the definition in the c-file is really defined and if it would push the compiler to
inline the call from remote_lrmd_ra.c. 
Usually this combination of 'extern inline' & 'inline'
is used with the definition in the header to inline where possible and still emit an 
external symbol afaik.